schemas/typescript/types.ts

```diff
@@ -112,86 +112,64 @@ export type QuotePreference =
  * remainder policy, use the object form.
  */
 export type FailureHandlingMode =
-  | "retry"
-  | "refund-instant"
+  | "refund-automatic"
   | "refund-claim"
   | "needs-new-signature";
 
-export type FailureHandling =
-  | FailureHandlingMode
-  | {
-      /** Whether partial execution may occur. Defaults to false if omitted. */
-      partialFill?: boolean;
-      /** How the missing portion is handled if a partial fill occurs. */
-      remainder: FailureHandlingMode;
-    };
-
 /**
  * Request for generating quotes
  * @description Request for generating quotes
  */
 export interface GetQuoteRequest {
   /** User requesting the quote */
   user: Address;
-  /** Available inputs for the quote. Order is significant if preference is 'input-priority' */
-  availableInputs: AvailableInput[];
-  /** Requested outputs for the quote */
-  requestedOutputs: RequestedOutput[];
-  /** Order type: 'swap-sell' = exact-input, 'swap-buy' = exact-output. If omitted, providers SHOULD assume 'swap-sell' for backward compatibility. */
-  orderType?: OrderType;
-  /** Minimum validity timestamp in seconds */
-  minValidUntil?: number;
-  /** Quote preference */
-  preference?: QuotePreference;
-  /**
-   * Explicit preference for submission responsibility and acceptable auth schemes.
-   * If provided, takes precedence over the legacy boolean.
-   */
-  originSubmission?: OriginSubmission;
-}
+  
+  intent: {
+    intentType: "oif-swap" ;
+    /** Available inputs for the quote. Order is significant if preference is 'input-priority' */
+    availableInputs: AvailableInput[];
+    /** Requested outputs for the quote */
+    requestedOutputs: RequestedOutput[];
+    /** Order type: 'swap-sell' = exact-input, 'swap-buy' = exact-output. If omitted, providers SHOULD assume 'swap-sell' for backward compatibility. */
+    orderType?: OrderType;
```

Comment:
When using exact-input (swap-sell), should we support slippage or minimum-output-amount parameters to make the operation safer for the user?

Reply:
for exact input flows, the output amount behaves as the "minimum output amount" . In exact output, the input amount behaves like "the maximum input amount"

Reply:
I see. 

It is bit confusing for understanding as it has double role.

We should then document it clearly.